### PR TITLE
Convert ChannelFragment to use Channels directly

### PR DESCRIPTION
**Pull request by kkong006**
Change Log:
- Convert ChannelFragment to use Channels
- Clicking on a Channel calls the PlayerService to play it
- Add ChannelsAdapter to show a Channel list

### CODE REVIEW

**Review by ttony678 — DISMISSED** [review]
Very clever use of ChannelsAdapter to re-use the recycler view using the Channel class.

File: app/src/main/java/appricottsoftware/clarity/fragments/ChannelFragment.java [hunk 1]
```diff
@@ -167,8 +156,12 @@ public void onClick(View v) {
...
 
-                            rListItemsSearch = new ArrayList<>();
-                            rAdapterSearch = new RecyclerAdapter(rListItemsSearch, getContext(), false);
+//                            rListItemsSearch = new ArrayList<>();
+//                            rAdapterSearch = new RecyclerAdapter(rListItemsSearch, getContext(), false);
+//                            channelRecycler.setAdapter(rAdapterSearch);
+
+                            searchChannels = new ArrayList<>();
+                            rAdapterSearch = new ChannelsAdapter(searchChannels, getContext(), false);
                             channelRecycler.setAdapter(rAdapterSearch);
```
Comment (lepatrick714) [inline app/src/main/java/appricottsoftware/clarity/fragments/ChannelFragment.java]: Could we get rid of these commented code? If not, please write a comment describing why the comments are necessary

File: app/src/main/java/appricottsoftware/clarity/fragments/ChannelFragment.java [hunk 2]
```diff
@@ -336,21 +314,19 @@ public void onFailure(int statusCode, Header[] headers, Throwable throwable, JSO
                 }
                 super.onFailure(statusCode, headers, throwable, errorResponse);
             }
+
             @Override
             public void onFailure(int statusCode, Header[] headers, Throwable throwable, JSONArray errorResponse) {
                 super.onFailure(statusCode, headers, throwable, errorResponse);
                 throwable.printStackTrace();
-                Log.e(TAG, "");
```
Comment (lepatrick714) [inline app/src/main/java/appricottsoftware/clarity/fragments/ChannelFragment.java]: We don't need a JSONArray errorResponse

File: app/src/main/java/appricottsoftware/clarity/fragments/ChannelFragment.java [hunk 3]
```diff
@@ -336,21 +314,19 @@ public void onFailure(int statusCode, Header[] headers, Throwable throwable, JSO
...
             }
 
             @Override
             public void onFailure(int statusCode, Header[] headers, String responseString, Throwable throwable) {
```
Comment (lepatrick714) [inline app/src/main/java/appricottsoftware/clarity/fragments/ChannelFragment.java]: We don't need a String responseString

File: app/src/main/java/appricottsoftware/clarity/fragments/ChannelFragment.java [hunk 4]
```diff
@@ -273,50 +262,39 @@ private void goToChannelList() {
...
-                    }
-                } catch (JSONException e) {
-                    e.printStackTrace();
-                }
-                rAdapter = new RecyclerAdapter(rListItems, getContext(), true);
-                channelRecycler.setAdapter(rAdapter);
             }
+
             @Override
             public void onSuccess(int statusCode, Header[] headers, String responseString) {
```
Comment (lepatrick714) [inline app/src/main/java/appricottsoftware/clarity/fragments/ChannelFragment.java]: We don't need a String responseString

**Review by lepatrick714 — CHANGES_REQUESTED** [review]

File: app/src/main/java/appricottsoftware/clarity/fragments/ChannelFragment.java (hunk 1 above)
Comment (kkong006) [reply app/src/main/java/appricottsoftware/clarity/fragments/ChannelFragment.java]: @mschaffrath I'm removing old code.

File: app/src/main/java/appricottsoftware/clarity/fragments/ChannelFragment.java (hunk 2 above)
Comment (kkong006) [reply app/src/main/java/appricottsoftware/clarity/fragments/ChannelFragment.java]: @mschaffrath I'm removing this block of code.

File: app/src/main/java/appricottsoftware/clarity/fragments/ChannelFragment.java (hunk 3 above)
Comment (kkong006) [reply app/src/main/java/appricottsoftware/clarity/fragments/ChannelFragment.java]: @mschaffrath I'm removing this block of code.

File: app/src/main/java/appricottsoftware/clarity/fragments/ChannelFragment.java (hunk 4 above)
Comment (kkong006) [reply app/src/main/java/appricottsoftware/clarity/fragments/ChannelFragment.java]: @mschaffrath I'm removing this block of code.

**Comment by kkong006** [comment]
@ttony678 That is the work of one of our lead software engineers, @mschaffrath! Please thank him.

@lepatrick714 I finished responding to the requested changes.

**Review by lepatrick714 — DISMISSED** [review]

**Comment by lepatrick714** [comment]
Please fix all merge conflicts before pulling this into master

**Comment by ttony678** [comment]
Lol are the merge conflicts from the change of putting context in constructor instead of as parameter in ClarityClient REST calls?

**Review by lepatrick714 — APPROVED** [review]

**Comment by lepatrick714** [comment]
Ship it!

**Pull request merged**